### PR TITLE
generate go client with latest openapi generator template

### PR DESCRIPTION
**Pull request by houqp**
Fix go client generation for openapi-generator 5.x. After this the client repo is updated, I will start a discussion on the dev list to finalize go client's release procedure following @msumit 's python runbook.

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

File: clients/gen/go.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ readonly CLEANUP_DIRS
 # shellcheck source=./clients/gen/common.sh
 source "${CLIENTS_GEN_DIR}/common.sh"
 
-VERSION=1.1.0
+VERSION=2.0.0
```
Comment (msumit) [inline clients/gen/go.sh]: Shouldn't it be `2.1.0`?

Comment (potiuk) [reply clients/gen/go.sh]: Hm. I remember we discussed versioning, but can't find it quickly. My memory is not the best recently (got some disruptions and distractions) but if I remember correctly, we agreed to only follow MAJOR release, assuming that most recent MAJOR client will work with Any MAJOR version of airflow (and in this case 2.0.0 is perfectly fine).

But I might be wrong - does anyone remember where we had that discussion and did we capture the result of it.?

Comment (potiuk) [reply clients/gen/go.sh]: @kaxil @ashb ?

Comment (msumit) [reply clients/gen/go.sh]: Nop, we decided to use Major.Minor release numbers from the Airflow and only last digit denoting fixes done on the client.

Comment (potiuk) [reply clients/gen/go.sh]: Ah. So my memory is failing me. Do you remember where the discussion was @msumit ? I think we need to capture it somewhere (unless it's already captured and I just did not find it :) ) ?

Comment (msumit) [reply clients/gen/go.sh]: @potiuk it was on the dev mailing list ([here](https://lists.apache.org/thread.html/r8b7f615f56eaa6d5c1955173295d70b47e038b92b84310a82e2765a9%40%3Cdev.airflow.apache.org%3E)).. I don't think that we've added it somewhere in the docs yet..

Comment (potiuk) [reply clients/gen/go.sh]: Thanks! Yeah. I remembered everything except the final conclusion :).

I created https://github.com/apache/airflow/pull/16422 with an attempt to capture all the SemVer rules we discussed so far. Happy to iterate and merge :)

**Review by potiuk — CHANGES_REQUESTED** [review]
Agree with @msumit -> version change is needed :)

**Comment by houqp** [comment]
@msumit @potiuk version updated.

**Review by msumit — APPROVED** [review]
lgtm

**Review by potiuk — APPROVED** [review]

**Pull request merged**